cogeo_mosaic/model.py

```diff
@@ -24,3 +31,169 @@ class MosaicJSON(BaseModel):
+    ):
+        """Create mosaic definition from features
+
+        Attributes
+        ----------
+        features: list[Dict], required
+            List of GeoJSON Features representing polygons of asset boundaries.
+        minzoom: int
+            Mosaic min-zoom.
+        maxzoom: int
+            Mosaic max-zoom.
+        quadkey_zoom: int, optional
+            Force quadkey zoom.
+        quiet: bool, optional (default: True)
+            Mask processing steps.
+
+        Returns
+        -------
+        mosaic_definition : dict
+            Mosaic definition.
+        """
+
+        quadkey_zoom = quadkey_zoom or minzoom
+
+        if not quiet:
+            click.echo(f"Get quadkey list for zoom: {quadkey_zoom}", err=True)
+
+        # Find dataset geometries
+        dataset_geoms = polygons(
+            [feat["geometry"]["coordinates"][0] for feat in features]
+        )
+        bounds = list(total_bounds(dataset_geoms))
+
+        tiles = burntiles.burn(features, quadkey_zoom)
+        tiles = [mercantile.Tile(*tile) for tile in tiles]
+
+        mosaic_definition = dict(
+            mosaicjson=version,
+            minzoom=minzoom,
+            maxzoom=maxzoom,
+            bounds=bounds,
+            center=[(bounds[0] + bounds[2]) / 2, (bounds[1] + bounds[3]) / 2, minzoom],
+            tiles={},
+            version="1.0.0",
+        )
+
+        if version == "0.0.2":
+            mosaic_definition.update(dict(quadkey_zoom=quadkey_zoom))
+
+        if not quiet:
+            click.echo(f"Feed Quadkey index", err=True)
+
+        # Create tree and find assets that overlap each tile
+        tree = STRtree(dataset_geoms)
+
+        for tile in tiles:
+            quadkey = mercantile.quadkey(tile)
+            tile_geom = polygons(mercantile.feature(tile)["geometry"]["coordinates"][0])
+
+            # Find intersections from rtree
+            intersections_idx = sorted(tree.query(tile_geom, predicate="intersects"))
+            if len(intersections_idx) == 0:
+                continue
+
+            intersections_geoms = [dataset_geoms[idx] for idx in intersections_idx]
+            intersections = [features[idx] for idx in intersections_idx]
+
+            dataset = [
+                {"identifier": accessor(f), "geometry": geom}
+                for (f, geom) in zip(intersections, intersections_geoms)
+            ]
+
+            if minimum_tile_cover is not None or tile_cover_sort:
+                dataset = _filter_and_sort(
+                    tile_geom,
+                    dataset,
+                    minimum_cover=minimum_tile_cover,
+                    sort_cover=tile_cover_sort,
+                )
+
+            if dataset and maximum_items_per_tile:
+                dataset = dataset[0:maximum_items_per_tile]
+
+            if dataset:
+                mosaic_definition["tiles"][quadkey] = [f["identifier"] for f in dataset]
+
+        return MosaicJSON.from_dict(mosaic_definition)
```

Comment:
I think going directly with `return cls(**mosaic_definition)` is fine here

Reply:
I was looking at the Geopandas source, and I got the `Classname(object)` syntax from them: https://github.com/geopandas/geopandas/blob/f13dafb881eaaf2bbf296e6f12189cc2c21a4216/geopandas/geodataframe.py#L387

Reply:
I prefer the `cls` constructor.  Regardless you don't need `from_dict`.

Reply:
Well it isn't _wrong_ at least, since it works